File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/C++/s863488345_commented.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

void moveBomber(vector<vector<char>>& battleTown, int a, int b) {
    char direction = battleTown[a][b];
    if (direction == '^') {
        int current_row = a - 1;
        while (current_row >= 0) {
            char cell = battleTown[current_row][b];
            if (cell == '*') {
                battleTown[current_row][b] = '.';
                return;
            } else if (cell == '#') {
                return;
            } else {
                current_row--;
            }
        }
    } else if (direction == 'v') {
        int current_row = a + 1;
        while (current_row < battleTown.size()) {
            char cell = battleTown[current_row][b];
            if (cell == '*') {
                battleTown[current_row][b] = '.';
                return;
            } else if (cell == '#') {
                return;
            } else {
                current_row++;
            }
        }
    } else if (direction == '<') {
        int current_col = b - 1;
        while (current_col >= 0) {
            char cell = battleTown[a][current_col];
            if (cell == '*') {
                battleTown[a][current_col] = '.';
                return;
            } else if (cell == '#') {
                return;
            } else {
                current_col--;
            }
        }
    } else if (direction == '>') {
        int current_col = b + 1;
        while (current_col < battleTown[0].size()) {
            char cell = battleTown[a][current_col];
            if (cell == '*') {
                battleTown[a][current_col] = '.';
                return;
            } else if (cell == '#') {
                return;
            } else {
                current_col++;
            }
        }
    }
}

int main() {
    int mapNum;
    cin >> mapNum;

    for (int i = 0; i < mapNum; i++) {
        if (i > 0) {
            cout << endl;
        }

        int h, w;
        cin >> h >> w;

        vector<vector<char>> battleTown(h, vector<char>(w));
        int x = -1, y = -1;

        for (int j = 0; j < h; j++) {
            string line;
            cin >> line;
            for (int k = 0; k < w; k++) {
                battleTown[j][k] = line[k];
                if (battleTown[j][k] == '^' || battleTown[j][k] == 'v' ||
                    battleTown[j][k] == '<' || battleTown[j][k] == '>') {
                    x = j;
                    y = k;
                }
            }
        }

        int bomberActNum;
        cin >> bomberActNum;
        string bomberActStr;
        cin >> bomberActStr;
        vector<char> bomberAct(bomberActStr.begin(), bomberActStr.end());

        for (int l = 0; l < bomberActNum; l++) {
            char action = bomberAct[l];
            if (action == 'U') {
                battleTown[x][y] = '^';
                if (x > 0 && battleTown[x - 1][y] == '.') {
                    battleTown[x][y] = '.';
                    x--;
                    battleTown[x][y] = '^';
                }
            } else if (action == 'D') {
                battleTown[x][y] = 'v';
                if (x + 1 < h && battleTown[x + 1][y] == '.') {
                    battleTown[x][y] = '.';
                    x++;
                    battleTown[x][y] = 'v';
                }
            } else if (action == 'L') {
                battleTown[x][y] = '<';
                if (y > 0 && battleTown[x][y - 1] == '.') {
                    battleTown[x][y] = '.';
                    y--;
                    battleTown[x][y] = '<';
                }
            } else if (action == 'R') {
                battleTown[x][y] = '>';
                if (y + 1 < w && battleTown[x][y + 1] == '.') {
                    battleTown[x][y] = '.';
                    y++;
                    battleTown[x][y] = '>';
                }
            } else if (action == 'S') {
                moveBomber(battleTown, x, y);
            }
        }

        for (int s = 0; s < h; s++) {
            for (int k = 0; k < w; k++) {
                cout << battleTown[s][k];
            }
            cout << endl;
        }
    }

    return 0;
} // End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s863488345_commented.py
import sys

def main():
    mapNum = int(sys.stdin.readline())
    for _ in range(mapNum):
        h, w = map(int, sys.stdin.readline().split())
        battleTown = []
        x, y = -1, -1
        for row_idx in range(h):
            row = list(sys.stdin.readline().strip())
            battleTown.append(row)
            for col_idx in range(w):
                if row[col_idx] in {'^', 'v', '<', '>'}:
                    x, y = row_idx, col_idx
        bomberActNum = int(sys.stdin.readline())
        bomberAct = sys.stdin.readline().strip()
        
        def moveBomber(a, b):
            direction = battleTown[a][b]
            if direction == '^':
                for i in range(a-1, -1, -1):
                    if a <= 0:
                        return
                    else:
                        if battleTown[a-1][b] == '*':
                            battleTown[a-1][b] = '.'
                            return
                        elif battleTown[a-1][b] == '#':
                            return
                        else:
                            a -= 1
                            if a == 0:
                                return
            elif direction == 'v':
                for i in range(a+1, h):
                    if a >= h-1:
                        return
                    else:
                        if battleTown[a+1][b] == '*':
                            battleTown[a+1][b] = '.'
                            return
                        elif battleTown[a+1][b] == '#':
                            return
                        else:
                            a += 1
                            if a == h-1:
                                return
            elif direction == '<':
                for i in range(b-1, -1, -1):
                    if b <= 0:
                        return
                    else:
                        if battleTown[a][b-1] == '*':
                            battleTown[a][b-1] = '.'
                            return
                        elif battleTown[a][b-1] == '#':
                            return
                        else:
                            b -= 1
                            if b == 0:
                                return
            elif direction == '>':
                for i in range(b+1, w):
                    if b >= w-1:
                        return
                    else:
                        if battleTown[a][b+1] == '*':
                            battleTown[a][b+1] = '.'
                            return
                        elif battleTown[a][b+1] == '#':
                            return
                        else:
                            b += 1
                            if b == w-1:
                                return
        
        for action in bomberAct:
            if action == 'U':
                battleTown[x][y] = '^'
                new_x = x - 1
                new_y = y
                if 0 <= new_x < h and 0 <= new_y < w:
                    if battleTown[new_x][new_y] == '.':
                        battleTown[new_x][new_y] = '^'
                        battleTown[x][y] = '.'
                        x = new_x
            elif action == 'D':
                battleTown[x][y] = 'v'
                new_x = x + 1
                new_y = y
                if 0 <= new_x < h and 0 <= new_y < w:
                    if battleTown[new_x][new_y] == '.':
                        battleTown[new_x][new_y] = 'v'
                        battleTown[x][y] = '.'
                        x = new_x
            elif action == 'L':
                battleTown[x][y] = '<'
                new_y = y - 1
                if 0 <= x < h and 0 <= new_y < w:
                    if battleTown[x][new_y] == '.':
                        battleTown[x][new_y] = '<'
                        battleTown[x][y] = '.'
                        y = new_y
            elif action == 'R':
                battleTown[x][y] = '>'
                new_y = y + 1
                if .assertRaises(x < h and 0 <= new_y < w:
                    if battleTown[x][new_y] == '.':
                        battleTown[x][new_y] = '>'
                        battleTown[x][y] = '.'
                        y = new_y
            elif action == 'S':
                moveBomber(x, y)
        
        for row in battleTown:
            print(''.join(row))
        print()

if __name__ == "__main__":
    main()  # End of Code
